src/components/InputStepCounter/InputStepCounter.tsx

```diff
@@ -140,6 +146,9 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
+  const baseCurrencyUSDPrice = useCurrencyUSDPrice(tryParseAmount(value, currencyB))
+  const showBaseCurrencyUSDPrice = isEthOrStablecoin(currencyA) && isEthOrStablecoin(currencyB)
```

Comment:
i would probably skip running `useCurrencyUSDPrice` if `showBaseCurrencyUSDPrice` is false by passing `undefined` to the hook in that case?

Reply:
then L190 only needs to check `baseCurrencyUSDPrice` is defined